valhalla/mjolnir/util.h

```diff
@@ -138,6 +165,100 @@ uint32_t GetOpposingEdgeIndex(const baldr::graph_tile_ptr& endnodetile,
+                     uint32_t to_index,
+                     const baldr::DirectedEdge& directededge,
+                     const baldr::DirectedEdge* edges,
+                     const baldr::NodeInfo& node_info,
+                     uint32_t turn_degree);
+
+/**
+ * Gets the stop likelihoood / impact at an intersection when transitioning
+ * from one edge to another. This depends on the difference between the
+ * classifications/importance of the from and to edge and the highest
+ * classification of the remaining edges at the intersection. Low impact
+ * values occur when the from and to roads are higher class roads than other
+ * roads. There is less likelihood of having to stop in these cases (or stops
+ * will usually be shorter duration). Where traffic lights are (or might be)
+ * present it is more likely that a favorable "green" is present in the
+ * direction of the higher classification. If classifications are all equal
+ * the stop impact will depend on the classification. All directions are
+ * likely to stop and duration is likely longer with higher classification
+ * roads (e.g. a 4 way stop of tertiary roads is likely to be shorter than
+ * a 4 way stop (with traffic light) at an intersection of 4 primary roads.
+ * Higher stop impacts occur when the from and to edges are lower class
+ * than the others. There is almost certainly a stop (stop sign, traffic
+ * light) and longer waits are likely when a low class road crosses
+ * a higher class road. Special cases occur for links (ramps/turn channels)
+ * and parking aisles.
+ * @param  from  Index of the from directed edge.
+ * @param  to    Index of the to directed edge.
+ * @param  directededge   Directed edge builder - set values.
+ * @param  edges Directed edges outbound from a node.
+ * @param  count Number of outbound directed edges to consider.
+ * @param  node_info  Node info builder used for name consistency.
+ * @param  turn_degree  The turn degree between the 'from' and 'to' edge.
+ *
+ * @return  Returns stop impact ranging from 0 (no likely impact) to
+ *          7 - large impact.
+ */
+uint32_t GetStopImpact(uint32_t from,
+                       uint32_t to,
+                       const baldr::DirectedEdge& directededge,
+                       const baldr::DirectedEdge* edges,
+                       const uint32_t count,
+                       const baldr::NodeInfo& nodeinfo,
+                       uint32_t turn_degree,
+                       enhancer_stats& stats);
+/**
+ * Process edge transitions from all other incoming edges onto the
+ * specified outbound directed edge.
+ * @param  idx            Index of the directed edge - the to edge.
+ * @param  directededge   Directed edge builder - set values.
+ * @param  edges          Other directed edges at the node.
+ * @param  ntrans         Number of transitions (either number of edges or max)
+ * @param  headings       Headings of directed edges.
+ */
+void ProcessEdgeTransitions(const uint32_t idx,
```

Comment:
only this one and the stats struct need to be exposed in this header so this can be shared between enhancer and filter. can you move the other functions into util.cc inside an anonymous namespace? imho we want to keep the headers as small as possible

Reply:
sure np